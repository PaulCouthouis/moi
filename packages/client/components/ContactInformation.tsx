import { ContactInformationEntity } from '@frontend/core'
import { x } from '@xstyled/styled-components'
import React from 'react'
import { SectionTitle } from './shared/SectionTitle'

export interface ContactInformationSerializeResult extends Omit<ContactInformationEntity, 'dateAvailable'> {
  dateAvailable: string
}

const IconLabelBlock: React.FC<{label: string, icon: string}> = ({ label, icon }) => {
  return (
    <x.div
      bg='#333'
      marginBottom='30px'
      p='20px 10px 15px'
      border='1px solid #444'
      display='flex'
      flexDirection='column'
      alignItems='center'
      w={{ _: '100%', md: '48%', xl: '24%' }}
    >
      <x.span className={`lnr ${icon}`} fontSize={40} color='#b5b5b5' />
      <x.p m='13px 0' fontSize={16} fontWeight={600} lineHeight='1.5em'>{label}</x.p>
    </x.div>
  )
}

export const ContactInformation: React.FC<{ contactInformation: ContactInformationSerializeResult}> = ({ contactInformation }) => {
  const { mainPhone, mainCity, mail, dateAvailable } = contactInformation
  return (
    <x.address id='contact' bg='#222' p='25px 30px 60px' marginTop='10px'>
      <SectionTitle>Contact</SectionTitle>
      <x.div display='flex' flexDirection={{ _: 'column', md: 'row' }} flexWrap='wrap' justifyContent='space-between'>
        <IconLabelBlock label={mail} icon='lnr-envelope' />
        <IconLabelBlock label={mainPhone} icon='lnr-smartphone' />
        <IconLabelBlock label={mainCity} icon='lnr-map-marker' />
        <IconLabelBlock label={`Disponible le ${new Date(dateAvailable).toLocaleDateString('fr-FR')}`} icon='lnr-power-switch' />
      </x.div>
    </x.address>
  )
}
