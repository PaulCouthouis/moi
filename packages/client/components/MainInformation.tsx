import { MainInformationEntity } from '@frontend/core'
import { x } from '@xstyled/styled-components'

export const MainInformation: React.FC<{ mainInformation: MainInformationEntity }> = (
  { mainInformation: { firstName, lastName, job, description } }
) => {
  return (
    <x.article
      w={{ sm: '65%' }}
      display='flex'
      alignItems='center'
      justifyContent='center'
      p='30px 30px 50px'
      fontFamily='Poppins, Helvetica, sans-serif'
      fontSize={14}
      lineHeight='1.85em'
    >
      <x.div maxWidth='100%'>
        <x.header color='#a5a5a5' fontWeight={300}>
          {job}
        </x.header>
        <x.h1 color='#f5f5f5' fontSize={36} fontWeight={600} lineHeight='1.2em' m='0 0 15px 0'>{firstName} {lastName}</x.h1>
        <x.p color='#d5d5d5' w='65ch' maxWidth='100%'> {description} </x.p>
      </x.div>
    </x.article>
  )
}
