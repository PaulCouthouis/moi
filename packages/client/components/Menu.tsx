import { x } from '@xstyled/styled-components'
import Image from 'next/image'
import { useCallback, useState } from 'react'
import styled from 'styled-components'
import profilePhoto from '../public/me.jpeg'

const Link = styled.a`
    display: block;
    padding: 10px 0;
    border-bottom: 1px solid #333;

    font-size: 14px;
    line-height: 1.85em;
    text-align: center;
    text-decoration: none;
    color: #f5f5f5;
`

export const Menu: React.FC<{ mainLabel: string }> = ({ mainLabel }) => {
  const [opened, setOpened] = useState(false)

  const handleMenuIconClick = useCallback(() => {
    setOpened(!opened)
  }, [opened])

  return (
    <x.nav position='fixed' w='100vw' zIndex='2' bg='#222' marginTop='-50px'>
      <x.header h='50px' p='0 20px' display='flex' alignItems='center' justifyContent='space-between' borderBottom='1px solid #444'>
        <x.div display='flex' alignItems='center'>
          <x.div h='34px' w='34px' overflowY='hidden'>
            <Image
              src={profilePhoto} alt='Photo de profile'
            />
          </x.div>
          <x.p marginLeft='8px' fontSize='18px' fontWeight={600}>{mainLabel}</x.p>
        </x.div>
        <x.button bg='#222' color='#fff' outline={{ focus: 'none' }} onClick={handleMenuIconClick}>
          <x.span className='lnr lnr-menu' fontSize='24px' fontWeight={900} />
        </x.button>
      </x.header>

      {opened &&
        <x.ul h='100vh' w='100vw' zIndex='2' bg='#222'>
          <li>
            <Link href='#' onClick={handleMenuIconClick}>A propos de moi</Link>
          </li>
          <li>
            <Link href='#method' onClick={handleMenuIconClick}>Ma m√©thodologie</Link>
          </li>
          <li>
            <Link href='#profile' onClick={handleMenuIconClick}>Profil</Link>
          </li>
          <li>
            <Link href='#contact' onClick={handleMenuIconClick}>Contact</Link>
          </li>
        </x.ul>}
    </x.nav>
  )
}
