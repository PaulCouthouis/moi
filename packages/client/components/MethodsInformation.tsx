import { MethodInformationEntity } from '@frontend/core/src'
import { x } from '@xstyled/styled-components'

export const MethodsInformation: React.FC<{
  methodsInformation: MethodInformationEntity[]
}> = ({ methodsInformation }) => {
  const articleItems = methodsInformation.map(({ title, description }, i) => (
    <x.article key={i}>
      <x.h3>{title}</x.h3>
      <x.p>{description}</x.p>
    </x.article>
  ))

  return (
    <x.section bg='#222' p='40px 30px 60px'>
      <x.h2 display='inline-block' marginBottom='20px' paddingBottom='3px' fontSize={21} fontWeight={600} borderBottom='2px solid #0099e5'>Ma MÃ©thodologie</x.h2>
      {articleItems}
    </x.section>
  )
}
