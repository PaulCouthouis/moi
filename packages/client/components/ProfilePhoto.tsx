import { x } from '@xstyled/styled-components'
import React from 'react'
import Image from 'next/image'
import profilePhoto from '../public/me.jpeg'

export const ProfilePhoto: React.FC = () => {
  return (
    <x.div h={{ _: 300, xs: 400, sm: 'inherit' }} overflowY='hidden'>
      <Image src={profilePhoto} alt='Photo de profile' />
    </x.div>
  )
}
