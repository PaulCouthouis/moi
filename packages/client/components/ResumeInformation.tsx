import { ExperiencesEntity, SkillEntity, TechnicalSkillsEntity } from '@frontend/core/src'
import { x } from '@xstyled/styled-components'
import { ResumeButton } from './shared/ResumeButton'
import { SectionTitle } from './shared/SectionTitle'

const ArticleTitle: React.FC = ({ children }) => {
  return (
    <x.h3
      display='inline-block'
      marginBottom='15px'
      paddingBottom='3px'
      fontSize={17}
      fontWeight={600}
      borderBottom='2px solid #0099e5'
    >
      {children}
    </x.h3>
  )
}

export const ResumeInformation: React.FC<{
  technicalSkills: TechnicalSkillsEntity[]
  experiences: ExperiencesEntity[]
}> = ({ technicalSkills, experiences }) => {
  const skillItem = (skills: SkillEntity[]): JSX.Element[] =>
    skills.map(({ name, value }, i) =>
      <x.div key={i} marginBottom='10px'>
        <x.div display='flex' justifyContent='space-between' lineHeight='1.25em'>
          <x.p fontSize={13} fontWeight={600}>{name}</x.p>
          <x.p fontSize={11} color='#aaa'>{value}%</x.p>
        </x.div>
        <x.div h='10px' border='1px solid #0099e5' borderRadius='10px'>
          <x.div h='6px' m='1px' backgroundColor='#0099e5' borderRadius='6px' w={`${value}%`} />
        </x.div>
      </x.div>
    )

  const skillsItems = technicalSkills.map(({ title, skills }, i) =>
    <x.article key={i} w='100%' marginBottom='60px'>
      <ArticleTitle>{title}</ArticleTitle>
      {skillItem(skills)}
    </x.article>
  )

  const experienceItems = experiences.map(({ startDate, endDate, company, title, description }, i) =>
    <x.section key={i} display='flex'>
      <x.div role='decoration' w='1px' background='#666' m={{ _: '0 24px 0 0', sm: '0' }} position={{ sm: 'relative' }} left='calc(17ch + 24.5px)'>
        <x.div role='decoration' w='11px' h='11px' m='7px 0 0 -5px' background='#222' border='2px solid #0099e5' borderRadius='5px' />
      </x.div>
      <x.div p='0 0 25px 0' display='flex' flexDirection={{ _: 'column', sm: 'row' }}>
        <x.header w='17ch' textAlign={{ sm: 'right' }} marginRight='50px'>
          <x.p p='3px 0' fontSize='14px' fontWeight={600} lineHeight='1.4em'>
            <time>{startDate}</time> - <time>{endDate}</time>
          </x.p>
          <x.p marginBottom='4px' color='#aaa' fontSize='12px' lineHeight='1.45em'>{company}</x.p>
        </x.header>
        <x.main w={{ sm: 'calc(100% - 17ch - 50px)' }}>
          <x.header marginBottom='3px' fontSize='16px' fontWeight={600} lineHeight='1.5em'> {title} </x.header>
          <x.p color='#d5d5d5' fontSize='12px' lineHeight='1.85em' maxWidth='65ch'>
            {description}
          </x.p>
        </x.main>
      </x.div>
    </x.section>
  )

  return (
    <x.div id='profile' bg='#222' p='25px 30px 60px' marginTop='10px'>
      <SectionTitle>Profil</SectionTitle>
      <x.main display='flex' flexWrap='wrap' justifyContent='space-around'>
        <x.article marginBottom='60px'>
          <ArticleTitle>Expériences</ArticleTitle>
          {experienceItems}
        </x.article>
        <x.div flex='1' minWidth='170px' maxWidth='710px'>
          {skillsItems}
          <x.div display='flex' justifyContent='center'>
            <ResumeButton>Plus de détails sur mon CV</ResumeButton>
          </x.div>
        </x.div>
      </x.main>
    </x.div>
  )
}
