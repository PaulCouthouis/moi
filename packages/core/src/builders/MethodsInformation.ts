import { Builder, DataBuilder, Repository } from './_builder'

export interface MethodInformationEntity {
  title: string
  description: string
}

export type MethodsInformationRepository = Repository<MethodInformationEntity[]>
export const MethodsInformationBuilder: Builder<MethodsInformationRepository, MethodInformationEntity[]> = DataBuilder
