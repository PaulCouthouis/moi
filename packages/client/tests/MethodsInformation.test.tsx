import renderer from 'react-test-renderer'
import { MethodsInformation } from '../components/MethodsInformation'

describe('Methods Information', () => {
  it('should return methods information of Paul Couthouis', () => {
    expect(renderer.create(
      <MethodsInformation
        methodsInformation={[
          {
            title: 'Test-Driven Development',
            description: 'Le TDD c\'est la meilleur manière de produire du code sans dette, sans bug, avec un design optimisé, une couverture de test complète'
          },
          {
            title: 'Hexagonal Architecture',
            description: 'Pour un produit très peu dépendant des choix techniques, l\'Hexagonal Architecture permet d\'isoler le code métier, soit le coeur de votre application. C\'est la garantie de faire évoluer votre produit dans le temps.'
          },
          {
            title: 'Design Fonctionnel',
            description: 'Un développeur n\'est pas seulement un codeur. Ensemble, conceptualisons la meilleur application en fonction des besoins réels du client'
          },
          {
            title: 'Intégration continue',
            description: 'Un des grands principes de l\'agilité, livrons notre application par petit bout de fonctionnalité et ainsi lui donner de la valeur.'
          }
        ]}
      />
    )).toMatchSnapshot()
  })
  it('should return methods information of John Doe', () => {
    expect(renderer.create(
      <MethodsInformation
        methodsInformation={[
          {
            title: 'Title 1',
            description: 'description 1'
          },
          {
            title: 'Title 2',
            description: 'description 2'
          }
        ]}
      />
    )).toMatchSnapshot()
  })
})
