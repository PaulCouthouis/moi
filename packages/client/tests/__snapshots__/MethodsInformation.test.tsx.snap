// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Methods Information should return methods information of John Doe 1`] = `
<section
  className="sc-bXexck jEqyfB"
>
  <h2
    className="sc-cBoqAE"
  >
    Mes méthodes pour projet réussi
  </h2>
  <article
    className="sc-eCApnc"
  >
    <h3
      className="sc-ciSkZP"
    >
      Title 1
    </h3>
    <p
      className="sc-eKYRIR"
    >
      description 1
    </p>
  </article>
  <article
    className="sc-eCApnc"
  >
    <h3
      className="sc-ciSkZP"
    >
      Title 2
    </h3>
    <p
      className="sc-eKYRIR"
    >
      description 2
    </p>
  </article>
</section>
`;

exports[`Methods Information should return methods information of Paul Couthouis 1`] = `
<section
  className="sc-bXexck jEqyfB"
>
  <h2
    className="sc-cBoqAE"
  >
    Mes méthodes pour projet réussi
  </h2>
  <article
    className="sc-eCApnc"
  >
    <h3
      className="sc-ciSkZP"
    >
      Test-Driven Development
    </h3>
    <p
      className="sc-eKYRIR"
    >
      Le TDD c'est la meilleur manière de produire du code sans dette, sans bug, avec un design optimisé, une couverture de test complète
    </p>
  </article>
  <article
    className="sc-eCApnc"
  >
    <h3
      className="sc-ciSkZP"
    >
      Hexagonal Architecture
    </h3>
    <p
      className="sc-eKYRIR"
    >
      Pour un produit très peu dépendant des choix techniques, l'Hexagonal Architecture permet d'isoler le code métier, soit le coeur de votre application. C'est la garantie de faire évoluer votre produit dans le temps.
    </p>
  </article>
  <article
    className="sc-eCApnc"
  >
    <h3
      className="sc-ciSkZP"
    >
      Design Fonctionnel
    </h3>
    <p
      className="sc-eKYRIR"
    >
      Un développeur n'est pas seulement un codeur. Ensemble, conceptualisons la meilleur application en fonction des besoins réels du client
    </p>
  </article>
  <article
    className="sc-eCApnc"
  >
    <h3
      className="sc-ciSkZP"
    >
      Intégration continue
    </h3>
    <p
      className="sc-eKYRIR"
    >
      Un des grands principes de l'agilité, livrons notre application par petit bout de fonctionnalité et ainsi lui donner de la valeur.
    </p>
  </article>
</section>
`;
